src/tsk_fs_file_handler.rs

```diff
@@ -0,0 +1,170 @@
+#![allow(non_camel_case_types)]
+
+use std::io::{Read, Seek, SeekFrom};
+use std::ffi::{CStr, CString};
+use super::{
+    tsk_fs_file::TskFsFile,
+    tsk_fs_attr::TskFsAttrType,
+    errors::TskError,
+    bindings as tsk
+};
+
+#[derive(Copy, Clone)]
+pub enum TskFsFileReadFlag {
+    None,
+    Slack,
+    NoID
+}
+
+#[derive(Copy, Clone)]
+pub struct TskFsFileHandler<'fs>{
+    tsk_fs_file: &'fs TskFsFile<'fs>,
+    _offset: i64,
+    attr_size: i64,
+    id: u16,
+    attr_to_read: TskFsAttrType,
+    pub read_flag: TskFsFileReadFlag
+}
+
+/// TskFsFileHandler struct is used to read file data.
+impl<'fs> TskFsFileHandler<'fs> {
+    /// Create TskFsFileHandler from TskFsFile. This will default to read $DATA attribute.
+    pub fn new(tsk_fs_file: &'fs TskFsFile<'fs>) -> Result<Self,TskError> {
+        let attr_size = (&tsk_fs_file.get_meta()?).size();
+        let mut attr_to_read: TskFsAttrType = TskFsAttrType::NTFS_DATA;
+        Ok( Self {
+            tsk_fs_file,
+            _offset: 0,
+            attr_size,
+            attr_to_read,
+            read_flag: TskFsFileReadFlag::NoID, // Ignore the id and read $DATA
+            id: 1
```

Comment:
A hard coded id will not work. There is no guarantee that the attribute to read has an id of 1.